vmm/src/vm.rs

```diff
@@ -1616,6 +1720,41 @@ impl<'a> Vm<'a> {
         // Unblock all CPU threads.
         vcpu_thread_barrier.wait();
 
+        let mut signals: [i32; MAX_SIGNAL_VAL] = [0; MAX_SIGNAL_VAL];
```

Comment:
I think replace this array with a Vec.

Reply:
In fact if you turned `DeviceManager.signal_present` which is currently an array of booleans into a Vec of ints then you could skip all this work and just pass a reference to that in.

Reply:
Well I even think we don't need this list of `present_signals`. As long as you check the signal number the event has been triggered on, and as long as `self.devices.console_input` is not `None`, then you can trigger the console size update.

Reply:
Done according to Sebastien's suggestion.